bookkeeper-server/src/main/java/org/apache/bookkeeper/http/DecommissionService.java

```diff
@@ -0,0 +1,103 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.http;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import org.apache.bookkeeper.client.BookKeeperAdmin;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.http.service.HttpService;
+import org.apache.bookkeeper.http.service.HttpServiceRequest;
+import org.apache.bookkeeper.http.service.HttpServiceResponse;
+import org.apache.bookkeeper.net.BookieSocketAddress;
+import org.apache.bookkeeper.util.JsonUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HttpService that handle Bookkeeper Configuration related http request.
+ */
+public class DecommissionService implements HttpService {
+
+    static final Logger LOG = LoggerFactory.getLogger(DecommissionService.class);
+
+    protected ServerConfiguration conf;
+
+    public DecommissionService(ServerConfiguration conf) {
+        Preconditions.checkNotNull(conf);
+        this.conf = conf;
+    }
+
+    /*
+     * decommission bookie,
+     */
+    @Override
+    public HttpServiceResponse handle(HttpServiceRequest request) throws Exception {
+        HttpServiceResponse response = new HttpServiceResponse();
+
+        if (HttpServer.Method.PUT == request.getMethod() || HttpServer.Method.POST == request.getMethod()) {
+            ClientConfiguration adminConf = new ClientConfiguration(conf);
+            BookKeeperAdmin admin = new BookKeeperAdmin(adminConf);
+            String requestBody = request.getBody();
+
+            if (requestBody == null) {
+                admin.close();
+                response.setCode(HttpServer.StatusCode.NOT_FOUND);
+                response.setBody("Null request body for DecommissionService.");
+                return response;
+            }
+
+            @SuppressWarnings("unchecked")
+            HashMap<String, String> configMap = JsonUtil.fromJson(requestBody, HashMap.class);
+            if (configMap != null && configMap.containsKey("bookie_src")) {
+                try {
+                    String bookieSrcString[] = configMap.get("bookie_src").split(":");
+                    BookieSocketAddress bookieSrc = new BookieSocketAddress(
+                      bookieSrcString[0], Integer.parseInt(bookieSrcString[1]));
+
+                    admin.decommissionBookie(bookieSrc);
+                    admin.close();
+                    response.setCode(HttpServer.StatusCode.OK);
+                    response.setBody("Success decommission Bookie " + bookieSrc.toString());
+                    return response;
+                } catch (Exception e) {
+                    admin.close();
+                    e.printStackTrace();
```

Comment:
remove

Reply:
Thanks. will change to LOG.error as Enrico suggested.